### ISSUE TITLE
picom: ../src/event.c:183: ev_create_notify: Assertion `ev->parent == ps->root' failed.

### ISSUE THREAD

**Issue by Safari77** [yshui/picom#371]
### Platform

Fedora 30 x86_64

### GPU, drivers, and screen setup
Radeon RX550, mesa-20.0.4

```
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: X.Org (0x1002)
    Device: Radeon 550 Series (POLARIS12, DRM 3.27.0, 4.19.113+, LLVM 9.0.1) (0x699f)
    Version: 20.0.4
    Accelerated: yes
    Video memory: 4096MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
Memory info (GL_ATI_meminfo):
    VBO free memory - total: 2770 MB, largest block: 2770 MB
    VBO free aux. memory - total: 4060 MB, largest block: 4060 MB
    Texture free memory - total: 2770 MB, largest block: 2770 MB
    Texture free aux. memory - total: 4060 MB, largest block: 4060 MB
    Renderbuffer free memory - total: 2770 MB, largest block: 2770 MB
    Renderbuffer free aux. memory - total: 4060 MB, largest block: 4060 MB
Memory info (GL_NVX_gpu_memory_info):
    Dedicated video memory: 4096 MB
    Total available memory: 8192 MB
    Currently available dedicated video memory: 2770 MB
OpenGL vendor string: X.Org
OpenGL renderer string: Radeon 550 Series (POLARIS12, DRM 3.27.0, 4.19.113+, LLVM 9.0.1)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 20.0.4
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 20.0.4
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 20.0.4
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

### Environment

Mate desktop 1.24, Xorg 1.20.6

### picom version

picom bdf809d0394396c058dddc54d13298c12ab15d4c
### Configuration:
```
picom --backend=glx --use-damage --vsync --experimental-backends
```

### Expected behavior
no crash

### Current Behavior
assert fail

### Stack trace
```
Program terminated with signal SIGABRT, Aborted.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	  return ret;
[Current thread is 1 (Thread 0x7fef90a9aec0 (LWP 2070556))]
Missing separate debuginfos, use: dnf debuginfo-install libconfig-1.7.2-3.fc30.x86_64 libev-4.25-8.fc30.x86_64
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007fef90e62895 in __GI_abort () at abort.c:79
#2  0x00007fef90e62769 in __assert_fail_base (fmt=0x7fef90fca688 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=0x55a493e209a9 "ev->parent == ps->root", file=0x55a493e2099a "../src/event.c", line=183, function=<optimized out>) at assert.c:92
#3  0x00007fef90e70566 in __GI___assert_fail (assertion=assertion@entry=0x55a493e209a9 "ev->parent == ps->root", 
    file=file@entry=0x55a493e2099a "../src/event.c", line=line@entry=183, 
    function=function@entry=0x55a493e20f80 <__PRETTY_FUNCTION__.54675> "ev_create_notify") at assert.c:101
#4  0x000055a493df9a9c in ev_create_notify (ev=0x55a494be94f0, ev=0x55a494be94f0, ps=0x55a49448d520) at ../src/utils.h:177
#5  ev_handle (ps=0x55a49448d520, ev=0x55a494be94f0) at ../src/event.c:712
#6  0x000055a493dd834d in x_event_callback (loop=<optimized out>, w=0x55a49448d520, revents=<optimized out>) at ../src/picom.c:1555
#7  0x00007fef9140b5e0 in ev_invoke_pending () from /usr/lib64/libev.so.4
#8  0x00007fef9140e793 in ev_run () from /usr/lib64/libev.so.4
#9  0x000055a493dd6d3a in session_run (ps=0x55a49448d520) at ../src/picom.c:2362
#10 main (argc=5, argv=0x7ffd9a0cc1a8) at ../src/picom.c:2463
(gdb) frame 4
#4  0x000055a493df9a9c in ev_create_notify (ev=0x55a494be94f0, ev=0x55a494be94f0, ps=0x55a49448d520) at ../src/utils.h:177
177			report_allocation_failure(func_name, file, line);
(gdb) p *ev
$1 = {response_type = 16 '\020', pad0 = 0 '\000', sequence = 14004, parent = 20972205, window = 20972209, x = 0, y = 0, width = 3840, height = 2160, 
  border_width = 0, override_redirect = 0 '\000', pad1 = 0 '\000'}
(gdb) p *ps
$2 = {xiow = {active = 1, pending = 0, priority = 0, data = 0x0, cb = 0x55a493dd8320 <x_event_callback>, next = 0x0, fd = 4, events = 1}, 
  unredir_timer = {active = 0, pending = 0, priority = 0, data = 0x0, cb = 0x55a493dd8cf0 <tmout_unredir_callback>, at = 0, repeat = 0}, 
  fade_timer = {active = 0, pending = 0, priority = 0, data = 0x0, cb = 0x55a493dd8ce0 <fade_timer_callback>, at = 0, repeat = 0}, 
  delayed_draw_timer = {active = 0, pending = 0, priority = 0, data = 0x0, cb = 0x55a493ddb070 <delayed_draw_timer_callback>, at = 0, repeat = 0}, 
  draw_idle = {active = 1, pending = 0, priority = 0, data = 0x0, cb = 0x55a493ddb0c0 <draw_callback>}, event_check = {active = 1, pending = 0, 
    priority = -2, data = 0x0, cb = 0x55a493dd83d0 <handle_queued_x_events>}, usr1_signal = {active = 1, pending = 0, priority = 0, data = 0x0, 
    cb = 0x55a493dd8570 <reset_enable>, next = 0x0, signum = 10}, int_signal = {active = 1, pending = 0, priority = 0, data = 0x0, 
    cb = 0x55a493dd86a0 <exit_enable>, next = 0x0, signum = 2}, backend_data = 0x55a4944ba430, backend_blur_context = 0x0, 
  drivers = (DRIVER_AMDGPU | DRIVER_RADEON), file_watch_handle = 0x55a4944a4de0, loop = 0x7fef91416720, server_grabbed = false, 
  dpy = 0x55a494478290, previous_xerror_handler = 0x7fef91309240 <_XDefaultError>, scr = 0, c = 0x55a49447a930, vis = 33, depth = 24, root = 1735, 
  root_height = 2160, root_width = 3840, overlay = 127, debug_window = 0, root_tile_fill = false, root_tile_paint = {pixmap = 0, pict = 0, 
    ptex = 0x0, fbcfg = 0x0}, root_image = 0x55a494be3740, screen_reg = {extents = {x1 = 0, y1 = 0, x2 = 3840, y2 = 2160}, data = 0x0}, 
  root_picture = 27262979, tgt_picture = 27262980, tgt_buffer = {pixmap = 0, pict = 0, ptex = 0x0, fbcfg = 0x0}, reg_win = 27262978, psglx = 0x0, 
  glx_prog_win = {prog = 0, unifm_opacity = -1, unifm_invert_color = -1, unifm_tex = -1, unifm_time = -1}, sync_fence = 0, first_frame = false, 
  root_flags = 0, o = {monitor_repaint = false, print_diagnostics = false, debug_mode = false, experimental_backends = true, write_pid_path = 0x0, 
    backend = BKEND_GLX, xrender_sync_fence = false, glx_no_stencil = false, glx_no_rebind_pixmap = false, glx_fshader_win_str = 0x0, 
    detect_rounded_corners = false, force_win_blend = false, resize_damage = 0, unredir_if_possible = false, unredir_if_possible_blacklist = 0x0, 
    unredir_if_possible_delay = 0, redirected_force = UNSET, stoppaint_force = UNSET, dbus = false, logpath = 0x0, benchmark = 0, benchmark_wid = 0, 
    paint_blacklist = 0x0, show_all_xerrors = false, no_x_selection = false, wintype_option = {{shadow = false, fade = false, focus = false, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = false, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = false, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}, {shadow = false, fade = false, focus = true, 
        full_shadow = false, redir_ignore = false, opacity = nan(0x8000000000000)}}, refresh_rate = 0, sw_opti = false, vsync = true, 
    vsync_use_glfinish = false, use_damage = true, shadow_red = 0, shadow_green = 0, shadow_blue = 0, shadow_radius = 18, shadow_offset_x = -15, 
    shadow_offset_y = -15, shadow_opacity = 0.75, shadow_exclude_reg_str = 0x0, shadow_blacklist = 0x0, shadow_ignore_shaped = false, 
    xinerama_shadow_crop = false, fade_in_step = 0.028000000000000001, fade_out_step = 0.029999999999999999, fade_delta = 10, 
    no_fading_openclose = false, no_fading_destroyed_argb = false, fade_blacklist = 0x0, inactive_opacity = 1, active_opacity = 1, 
    inactive_opacity_override = false, frame_opacity = 1, detect_client_opacity = false, blur_method = BLUR_METHOD_NONE, blur_radius = 3, 
    blur_deviation = 0.84089641999999998, blur_background_frame = false, blur_background_fixed = false, blur_background_blacklist = 0x0, 
    blur_kerns = 0x0, blur_kernel_count = 0, inactive_dim = 0, inactive_dim_fixed = false, invert_color_list = 0x0, opacity_rules = 0x0, 
    max_brightness = 1, mark_wmwin_focused = false, mark_ovredir_focused = false, use_ewmh_active_win = false, focus_blacklist = 0x0, 
    detect_transient = false, detect_client_leader = false, track_leader = false, no_ewmh_fullscreen = false, transparent_clipping = false}, 
  tmout_unredir_hit = false, redraw_needed = true, damage = 0x55a494bdef10, damage_ring = 0x55a494bdef10, ndamage = 5, redirected = true, 
  alpha_picts = 0x0, fade_time = 0, ignore_head = 0x0, ignore_tail = 0x55a49448da18, blur_kerns_cache = 0x0, quit = false, pending_updates = false, 
  expose_rects = 0x55a4944b1360, size_expose = 1, n_expose = 0, windows = 0x55a49449e210, window_stack = {next = 0x55a494c23648, 
    prev = 0x55a49449e248}, active_win = 0x55a4944adf50, active_leader = 0, black_picture = 0, cshadow_picture = 0, white_picture = 0, 
  gaussian_map = 0x55a494490c40, shadow_exclude_reg = {extents = {x1 = 0, y1 = 0, x2 = 0, y2 = 0}, 
    data = 0x7fef91259c80 <pixman_region32_empty_data_>}, refresh_rate = 0, refresh_intv = 0, paint_tm_offset = 0, xfixes_event = 86, 
  xfixes_error = 138, damage_event = 90, damage_error = 150, render_event = 0, render_error = 140, composite_event = 0, composite_error = 0, 
  composite_opcode = 141, shape_exists = true, shape_event = 64, shape_error = 0, randr_exists = true, randr_event = 88, randr_error = 145, 
  present_exists = true, glx_exists = true, glx_event = 94, glx_error = 156, xinerama_exists = false, xinerama_scr_regs = 0x0, xinerama_nscrs = 0, 
  xsync_exists = true, xsync_event = 83, xsync_error = 134, xrfilter_convolution_exists = false, atoms = 0x55a49448e100, atoms_wintypes = {0, 355, 
    374, 372, 339, 343, 373, 336, 340, 338, 341, 342, 396, 335, 337}, track_atom_lst = 0x0, dbus_data = 0x0, vsync_wait = 0x0}
```

**Comment by tryone144** [comment]
Can confirm this is related to https://github.com/yshui/picom/commit/bdf809d0394396c058dddc54d13298c12ab15d4c. Previous commits don't exhibit this behaviour.

I can reliably reproduce the issue by opening the context-menu (right-click) or native drop-down inputs in chrome under i3.
- chromium 80.0.3987.163-1
- i3-gaps 4.18-1

If specific configs/logs are required, I'll provide them.

**Comment by yshui** [comment]
Yes, they were the broken fix for #299, I am about to revert them.
